src/cpp/src/continuous_batching/pipeline_base.cpp

```diff
@@ -168,34 +161,29 @@ ContinuousBatchingPipeline::IContinuousBatchingPipeline::generate(
 
     std::vector<ov::Tensor> input_embeds_list;
     std::vector<VLMPerfMetrics> vlm_perf_metrics(prompts.size());
+    std::vector<EncodedImage> encoded_images = {};
 
     if (m_is_chat_conversation) {
         OPENVINO_ASSERT(1 == prompts.size(), "Can't chat with multiple prompts");
         const auto& rgbs = rgbs_vector[0];
-        auto prompt_with_tags = prompts[0];
-        if (!m_inputs_embedder->prompt_has_image_tag(prompt_with_tags)) {
-            prompt_with_tags = add_image_tags_to_prompt(prompt_with_tags, rgbs, m_history_images.size());
-        }
-        m_history.push_back({{"role", "user"}, {"content", prompt_with_tags}});
+        const auto prompt = prompts[0];
+        const auto [unified_prompt, image_sequence] = m_inputs_embedder->normalize_prompt(prompt, m_image_id, rgbs.size());
```

Comment:
`rgbs` maybe batched. `encoded_images` should be used

Reply:
Done.